diff --git a/src/Service/PromotionsService.php b/src/Service/PromotionsService.php
index 2f14383..bbb46de 100644
--- a/src/Service/PromotionsService.php
+++ b/src/Service/PromotionsService.php
@@ -43,17 +43,15 @@ class PromotionsService extends AbstractService
                     $this->getOffset($limit, $page)
                 );
 
-                $promotions = $this->mapManyEntities($dbEntities);
-                return $this->groupPromotionsByType($promotions);
+                return $this->mapManyEntities($dbEntities);
             }
         );
     }
 
-    /**
-     * @param Promotion[] $promotions
-     */
-    public function groupPromotionsByType(array $promotions): array
+    public function findActivePromotionsByEntityGropuedByType()
     {
+        $promotions = $this->findActivePromotionsByEntity();
+
         $groupedPromotions = [
             'regular' => [],
             'super' => [],
